### DISCUSSION TITLE
openMVG_main_IncrementalSfM

### DISCUSSION THREAD

**Discussion by Sabney09** [openMVG/openMVG#2381] [General]
Was this target deprecated?  or name changed?  It doesn't seem to build from:

`$ make -j$(nproc)`

 Here is my output from build:
 cmake --build . --target help
The following are some of the valid targets for this Makefile:
... all (the default if no target is provided)
... clean
... depend
... edit_cache
... install
... install/local
... install/strip
... list_install_components
... rebuild_cache
... arrow_cpp
... check
... openMVG_main_AlternativeVO_autogen
... openMVG_main_AlternativeVO_autogen_timestamp_deps
... openMVG_sample_describe_and_match_GUI_autogen
... openMVG_sample_describe_and_match_GUI_autogen_timestamp_deps
... openMVG_sample_image_undistort_autogen
... openMVG_sample_image_undistort_autogen_timestamp_deps
... uninstall
... domset
... lib_CoinUtils
... lib_Osi
... lib_OsiClpSolver
... lib_clp
... main_svgSample
... openMVG_easyexif
... openMVG_exif
... openMVG_fast
... openMVG_features
... openMVG_geometry
... openMVG_image
... openMVG_kvld
... openMVG_lInftyComputerVision
... openMVG_linearProgramming
... openMVG_main_AlternativeVO
... openMVG_main_ChangeLocalOrigin
... openMVG_main_ColHarmonize
... openMVG_main_ComputeClusters
... openMVG_main_ComputeFeatures
... openMVG_main_ComputeMatches
... openMVG_main_ComputeSfM_DataColor
... openMVG_main_ComputeStructureFromKnownPoses
... openMVG_main_ComputeVLAD
... openMVG_main_ConvertList
... openMVG_main_ConvertMatches
... openMVG_main_ConvertSfM_DataFormat
... openMVG_main_ExportCameraFrustums
... openMVG_main_ExportUndistortedImages
... openMVG_main_FrustumFiltering
... openMVG_main_GeometricFilter
... openMVG_main_ListMatchingPairs
... openMVG_main_MatchesToTracks
... openMVG_main_MatchingRetrievalQuality
... openMVG_main_PairGenerator
... openMVG_main_PointsFiltering
... openMVG_main_SfM
... openMVG_main_SfMInit_ImageListing
... openMVG_main_SfMInit_ImageListingFromKnownPoses
... openMVG_main_SfM_Localization
... openMVG_main_SplitMatchFileIntoMatchFiles
... openMVG_main_VO
... openMVG_main_benchANN
... openMVG_main_evalQuality
... openMVG_main_exif_Parsing
... openMVG_main_exportKeypoints
... openMVG_main_exportMatches
... openMVG_main_exportTracks
... openMVG_main_geodesy_registration_to_gps_position
... openMVG_main_openMVG2Agisoft
... openMVG_main_openMVG2CMPMVS
... openMVG_main_openMVG2Colmap
... openMVG_main_openMVG2MESHLAB
... openMVG_main_openMVG2MVE2
... openMVG_main_openMVG2MVSTEXTURING
... openMVG_main_openMVG2NVM
... openMVG_main_openMVG2PMVS
... openMVG_main_openMVG2WebGL
... openMVG_main_openMVG2openMVS
... openMVG_main_openMVGSpherical2Cubic
... openMVG_matching
... openMVG_matching_image_collection
... openMVG_multiview
... openMVG_multiview_test_data
... openMVG_numeric
... openMVG_robust_estimation
... openMVG_sample_cameras_undistoBrown
... openMVG_sample_describe_and_match_GUI
... openMVG_sample_features_affine
... openMVG_sample_features_kvld
... openMVG_sample_features_siftPutative
... openMVG_sample_geodesy_show_exif_gps_position
... openMVG_sample_image_matching
... openMVG_sample_image_matching_gmsfilter
... openMVG_sample_image_undistort
... openMVG_sample_main_features_repeatability_dataset
... openMVG_sample_multiview_robustEssential
... openMVG_sample_multiview_robustEssential_ba
... openMVG_sample_multiview_robustEssential_spherical
... openMVG_sample_multiview_robustFundamental
... openMVG_sample_multiview_robustFundamental_guided
... openMVG_sample_multiview_robustHomography
... openMVG_sample_multiview_robustHomography_guided
... openMVG_sample_pano_converter
... openMVG_sample_pano_spherical_to_cubic
... openMVG_sample_robust_estimation_Homography
... openMVG_sample_visualize_sfm_data
... openMVG_sfm
... openMVG_stlplus
... openMVG_system
... rerun_sdk
... vlsift

**Comment by pmoulon** [comment]
The new target is openMVG_main_sfm it is the front end for all sfm versions

**Reply by Sabney09** [reply]
OK.. thanks
i seem to be stuck with an error no matter what i do... I updated the command to the new "openMVG_main_sfm" - but i'm still stuck:

command line output:

`ERROR: [main_SfM.cpp:466] Cannot load the match file.`
'Command failed: Command '/usr/local/bin/openMVG_main_SfM -i /route_to/sfm_data.json -o /route_to/sparse -m /route_to/openMVG_Build/workspace/matches -M /route_to/openMVG_Build/workspace/matches/matches.putative.bin --sfm_engine INCREMENTAL -a 1.jpg -b 2.jpg -c 3 --triangulation_method 3 --resection_method 4' returned non-zero exit status 1.`

what causes the openMVG_main_SfM command to "fail"?  

what's interesting is "cannot load"  - the 'matches.putative.bin' file is valid, it has data, it seems to be in the right format...its in the right path...

**Reply by pmoulon** [reply]
You need to run the matching pipeline first. Take a look to the tutorial py file that is a long the main_sfm cpp file https://github.com/openMVG/openMVG/wiki/OpenMVG-on-your-image-dataset
